ibc-primitives/src/types/timestamp.rs

```diff
@@ -268,12 +285,12 @@ impl scale_info::TypeInfo for Timestamp {
 
 #[derive(Debug, Display, derive_more::From)]
 pub enum TimestampError {
-    /// parse int error: {0}
-    ParseInt(ParseIntError),
-    /// try from int error: {0}
-    TryFromInt(TryFromIntError),
-    /// failed to convert timestamp: {0}
-    Conversion(time::error::ComponentRange),
+    /// failed to parse integer error: {0}
```

Comment:
Additionally along with this PR, I've addressed a few inconsistency in `TimestampError` with our new error handling convention.

Reply:
A few nits:

```
failed to parse integer: {0}
failed try_from on integer: {0}
failed to convert offset date
```

Reply:
[4f11258](https://github.com/cosmos/ibc-rs/pull/1353/commits/4f112581c4155194e8830db65f85e14355e6db03)